prover/prover.go

```diff
@@ -677,3 +682,51 @@ func (p *Prover) cancelProof(ctx context.Context, blockID uint64) {
 		delete(p.currentBlocksBeingProven, blockID)
 	}
 }
+
+func (p *Prover) checkProofWindowsExpired(ctx context.Context) error {
+	for blockId, cancel := range p.currentBlocksWaitingForProofWindow {
+		if err := p.checkProofWindowExpired(ctx, blockId, cancel); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (p *Prover) checkProofWindowExpired(ctx context.Context, blockId uint64, cancel cancelFunc) error {
+	p.currentBlocksWaitingForProofWindowMutex.Lock()
+	defer p.currentBlocksWaitingForProofWindowMutex.Unlock()
```

Comment:
this this `Unlock` will prevent the `cancel` from running since it also waits for a Lock to delete the entry from the map. most likely better to remove the `cancel` func entirely and just delete from the map in this function.